<launch>
   <arg name="frame_id_global"                              default="map"/>
   <arg name="frame_id_robot_base"                          default="base_footprint"/>
   <arg name="topic_goal"                                   default="/move_base_simple/goal"/>
   <arg name="potential_field_rows"                         default="120"/>
   <arg name="potential_field_cols"                         default="120"/>
   <arg name="potential_field_resolution"                   default="0.05"/>
   <arg name="weight_attraction_field"                      default="0.1"/>
   <arg name="weight_repulsion_field"                       default="0.1"/>
   <arg name="distance_threshold_repulsion_field"           default="0.5"/>
   <arg name="max_path_length"                              default="6.0"/>
   <arg name="path_search_range"                            default="1"/>
   <arg name="weight_potential_field"                       default="1.0"/>
   <arg name="weight_angle"                                 default="0.2"/>
   <arg name="collision_count_to_replanning"                default="3"/>
   <arg name="hit_distance_to_replanning"                   default="0.1"/>

   
   <node pkg="potbot_pathplanner" name="pathplanner_pp" type="potbot_pathplanner_node">
      <param name="frame_id_global"                         value="$(arg frame_id_global)"/>
      <param name="frame_id_robot_base"                     value="$(arg frame_id_robot_base)"/>
      <remap from="goal"                                    to="$(arg topic_goal)"/>
      <param name="potential_field_rows"                    value="$(arg potential_field_rows)"/>
      <param name="potential_field_cols"                    value="$(arg potential_field_cols)"/>
      <param name="potential_field_resolution"              value="$(arg potential_field_resolution)"/>
      <param name="weight_attraction_field"                 value="$(arg weight_attraction_field)"/>
      <param name="weight_repulsion_field"                  value="$(arg weight_repulsion_field)"/>
      <param name="distance_threshold_repulsion_field"      value="$(arg distance_threshold_repulsion_field)"/>
      <param name="max_path_length"                         value="$(arg max_path_length)"/>
      <param name="path_search_range"                       value="$(arg path_search_range)"/>
      <param name="weight_potential_field"                  value="$(arg weight_potential_field)"/>
      <param name="weight_angle"                            value="$(arg weight_angle)"/>
      <param name="collision_count_to_replanning"           value="$(arg collision_count_to_replanning)"/>
      <param name="hit_distance_to_replanning"              value="$(arg hit_distance_to_replanning)"/>
   </node>
   
</launch>
